File: backend/job-service/src/main/java/com/example/demo/job/domain/Job.java
package com.example.demo.job.domain;

import com.example.demo.job.domain.Employer;
import com.example.demo.job.domain.JobSkill;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "jobs")
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false, fetch = jakarta.persistence.FetchType.LAZY)
    @JoinColumn(name = "employer_id", nullable = false)
    private Employer employer;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    private String location;

    private String seniority;

    @Column(name = "work_type")
    private String workType;

    private Integer salaryMin;

    private Integer salaryMax;

    @OneToMany(mappedBy = "job", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<JobSkill> jobSkills = new ArrayList<>();

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;

    public Job() {
        // Default constructor for JPA
    }

    public Job(Employer employer, String title, String description, String location, String workType,
               String seniority, Integer salaryMin, Integer salaryMax) {
        this.employer = employer;
        this.title = title;
        this.description = description;
        this.location = location;
        this.workType = workType;
        this.seniority = seniority;
        this.salaryMin = salaryMin;
        this.salaryMax = salaryMax;
    }

    @PrePersist
    public void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = Instant.now();
    }

    public Long getId() {
        return id;
    }

    public Employer getEmployer() {
        return employer;
    }

    public void setEmployer(Employer employer) {
        this.employer = employer;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getSeniority() {
        return seniority;
    }

    public void setSeniority(String seniority) {
        this.seniority = seniority;
    }

    public String getWorkType() {
        return workType;
    }

    public void setWorkType(String workType) {
        this.workType = workType;
    }

    public Integer getSalaryMin() {
        return salaryMin;
    }

    public void setSalaryMin(Integer salaryMin) {
        this.salaryMin = salaryMin;
    }

    public Integer getSalaryMax() {
        return salaryMax;
    }

    public void setSalaryMax(Integer salaryMax) {
        this.salaryMax = salaryMax;
    }

    public List<JobSkill> getJobSkills() {
        return jobSkills;
    }

    public void addJobSkill(JobSkill jobSkill) {
        jobSkills.add(jobSkill);
        jobSkill.setJob(this);
    }

    public void removeJobSkill(JobSkill jobSkill) {
        jobSkills.remove(jobSkill);
        jobSkill.setJob(null);
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }
}

File: frontend/src/types/job.ts
export type WorkType = "ONSITE" | "HYBRID" | "REMOTE";

export type Seniority = "INTERN" | "JUNIOR" | "MID" | "SENIOR" | "LEAD";

export interface Employer {
  id: number;
  name: string;
  website?: string;
  description?: string;
  location?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateEmployerRequest {
  name: string;
  website?: string;
  description?: string;
  location?: string;
}

export interface Job {
  id: number;
  employer?: Employer;
  employerId?: number;
  company?: string;
  companyName?: string;
  title: string;
  description: string;
  location: string;
  salaryMin?: number;
  salaryMax?: number;
  workType?: WorkType;
  seniority?: Seniority;
  skills?: string[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateJobRequest {
  employerId: number;
  title: string;
  description: string;
  location: string;
  workType: WorkType;
  seniority: Seniority;
  salaryMin?: number;
  salaryMax?: number;
  skills: string[];
}

export interface UpdateJobRequest {
  title?: string;
  description?: string;
  location?: string;
  workType?: WorkType;
  seniority?: Seniority;
  salaryMin?: number;
  salaryMax?: number;
  skills?: string[];
}
